tests/integration_tests/test_examples.py

```diff
@@ -1,39 +1,48 @@
+"""This is an integration test to make sure scripts from examples/
+work when executing `./examples/**/*.py`.
 """
-This is an integration test to make sure scripts from examples/
-work when running `python examples/xx/xxx.py`.
-"""
-from garage.experiment import LocalRunner, run_experiment
-from garage.np.baselines import LinearFeatureBaseline
-from garage.tf.algos import VPG
-from garage.tf.envs import TfEnv
-from garage.tf.policies import CategoricalMLPPolicy
+import os
+from pathlib import Path
+import subprocess
+
+import pytest
+
+EXAMPLES_ROOT_DIR = Path('examples/')
+
+
+def enumerate_algo_examples():
+    """Return a list of paths for all algo examples
+
+    Returns:
+        List[str]: list of path strings
+
+    """
+    return map(str, EXAMPLES_ROOT_DIR.glob('*/**/*.py'))
 
 
-def _run_task(snapshot_config, *_):
-    with LocalRunner(snapshot_config=snapshot_config) as runner:
-        env = TfEnv(env_name='CartPole-v1')
+@pytest.mark.parametrize('filepath', enumerate_algo_examples())
+def test_algo_examples(filepath):
+    """Test algo examples.
 
-        policy = CategoricalMLPPolicy(name='policy',
-                                      env_spec=env.spec,
-                                      hidden_sizes=(32, 32))
+    Args:
+        filepath (str): path string of example
 
-        baseline = LinearFeatureBaseline(env_spec=env.spec)
+    """
+    env = os.environ.copy()
+    env['GARAGE_EXAMPLE_TEST_N_EPOCHS'] = '1'
+    assert subprocess.run([filepath, '--n_epochs', '1'], check=True,
+                          env=env).returncode == 0
```

Comment:
doesn't `check=True` have the same affect as this assert? AFAICT you will never fail this assert because `subprocess.run` with raise a CalledProcessError if the process fails.

Have you verified that this breaks when an example is broken?

Reply:
One of the examples I removed in another PR had an ImportError and the test for that. I tried that on an example now and pytest shows a failure, although it's not an AssertionError, but a CalledProcessError. So it never reached the assertion. Without check=True, it works fine and throws AssertionError. 

But another thing I found while checking this is that for something like ValueError, the test still passes, and if it's a NameError, the test just hangs. If I do sys.exit(1), I assumed it would explicitly set the return code but the assertion still passed. These things are with or without check=True :confused:

Reply:
The experiment wrapper is swallowing exceptions, and it doesn't return return code of the experiment process:
https://github.com/rlworkgroup/garage/blob/master/src/garage/experiment/experiment.py#L261

I think you should change 264-265 to just `raise` which re-raises the exception after printing it. I also think you need to either raise an exception if `subprocess.call` doesn't return 0 or you can do:
```
# experiment.py
return subprocess.call(...)

# launcher
return_code = run_experiment(...)
sys.exit(return_code)
```

I think raising an exception is more natural.

Seems like everything would be fixed if you changed experiment.py to use `subprocess.run(..., check=True)` and stop swallowing the exceptions.

Reply:
Setting timeout parameter on subprocess.run could be one possible solution to this but might be flaky

Reply:
Ah okay. I'll try that.

Reply:
That worked for ValueErrors, NameError still hangs and doesn't seem to be reaching the except block in experiment.py:263

Reply:
is the NameError raised in the launcher file or in the algorithm itself?

Reply:
In the launcher. From this line: https://github.com/rlworkgroup/garage/blob/d4c2d8711c205d247d2106dc02b976f79171b319/src/garage/experiment/experiment_wrapper.py#L165, control goes to run_task, error is raised there, and control doesn't return.

Reply:
hmm rather than going down this rabbit hole (which is ultimately caused by bad code we're trying to delete), perhaps just install this and set a timeout for the test: https://pypi.org/project/pytest-timeout/

Reply:
Please add a comment along the lines of
> Don't use check=True, since that causes test to hang on failure.

Reply:
This check=True was not the cause of hangs, this one caused the test to exit with an Error before the assertion was evaluated. Adding that in comment. The hang was because check was False in experiment.py, and I guess it makes sense there by looking at it since it's in a try except, so we definitely want it to throw error on failure. I can add a comment there too if you think it's helpful.

Reply:
Ah, I see. I don't really think we need a comment then, but feel free to use your best judgement.